### ISSUE TITLE
Significant Lag, Forced Windowed Mode doesnt work at all, Gitches w/ Textures and Shadows

### ISSUE THREAD

**Issue by retconn** [elishacloud/Silent-Hill-2-Enhancements#57]
I downloaded the lasted HD enhancement. I am having more issues than a previous version I had (don't know what version of enhancements there were exactly). The issues I am having are not addressed or solved from the troubleshoot section of your website.
(All my current config settings are at the bottom of my post)

 1. Game lag. I tried playing with no HD textures, no noise filters or complex fog and still significant game lag in certain outdoor environments. I also decreased settings significantly in the ini config file for nemesis complex fog fix and noise filter. FPS is set at 30. Also game lag in certain scenes like meeting Angela in the apartments with the mirror. Side note the song that plays there stops looping after Angela leaves and the picture on the wall, the texture glitches and flickers. 
2. Force Windowed mode does not work at all. I have it set in the config file (see below) but every time it uses full screen and no options to make it windowed. 
 3. The first scene were Maria dies and James sits in the elevator, his hair texture freezes several times and looks strange
 4. Shadow effect in the scene where you have played the tape in hotel and talking to Laura, the shadow flickers oddly and is distracting.
 5. Only have experienced with In The Water ending the audio where Mary is narrating her letter the audio will drop in and out and specific points where its really noticeable. 

I dont have anti aliasing set for the game on my GPU, and all my operations seem fine, everything is low like CPU, GPU, Disk. My memory is at about 35% playing the game when I am using OBS so I wouldnt think it is that... 

If you could help out that would be greatly appreciated thank you!﻿

;; Config file for Silent Hill 2 Enhancements

[General]
AutoUpdateModule = 1 // Checks for updated versions of the SH2 Enhancement module.
d3d8to9 = 1 // Causes Silent Hill 2 to use DirectX 9.
UseCustomModFolder = 1 // Uses files in the 'sh2e' folder rather than the 'data' folder.
WidescreenFix = 1 // Enables WidescreenFixesPack and sh2proxy module. See options below.

[GAME FIXES]
CemeteryLightingFix = 1 // Prevents saving wrong lighting values in the cemetery, which can corrupt the fog.
CatacombsMeatRoomFix = 1 // Updates the color and lighting of the catacomb meat cold rooms to be more like the PS2.  This update also requires updated 'ps189.map' and 'ps193.map' files.
DisableRedCrossInCutScenes = 1 // Hides the low health indicator during in-game cutscenes. Use with FastTransitions = 1 for best results.
EnableSFXAddrHack = 1 // Allows custom SFX files to be used. Required if using the Audio Enhancement Pack.
IncreaseDrawDistance = 1 // Increases the front draw distance in certain areas to hide the environment loading in.
Nemesis2000FogFix = 1 // Enables the Nemesis2000 Fog Fix module. See options below.
NoCDPatch = 1 // Removes the CD checks on some versions of SH2.
PS2StyleNoiseFilter = 1 // Adjusts noise filter to resemble the PS2 version. Use with IncreaseNoiseEffectRes = 512 for best results.
ResetScreenRes = 1 // Fixes an issue with WineD3D where the screen stays black after exiting SH2.
RowboatAnimationFix = 1 //  Fixes an issue with rowboat animation if you exit to the menu and reload the game.

[Plugins]
LoadPlugins = 0 // Loads ASI plugins
LoadFromScriptsOnly = 0 // Loads ASI plugins only from the 'scripts' and 'plugins' folders.

[WINDOWED MODE]
EnableWndMode = 1 // Enables forced windowed mode.
WndModeBorder = 1 // Enables window border when WndMode is enabled.

// ThirteenAG Silent Hill 2 WidescreenFixesPack
[MAIN]
ResX = 0 // Resolution width, 0 for default
ResY = 0 // Resolution height, 0 for default
FMVWidescreenMode = 1 // Cutscenes and movies will display better in widescreen.
FMVWidescreenEnhancementPackCompatibility = 1 // Adds compatibility when using the FMV Enhancement Pack.
Fix2D = 1 // Enable to set 2D images to their original full screen aspect ratio. Disable to stretch the images to fill the screen.

// Some features from https://github.com/emoose/sh2proxy
[MISC]
DisableCutsceneBorders = 1 // Removes cutscene letterboxing.
SingleCoreAffinity = 1 // Limits game to one CPU core to prevent sound loop and other various bugs.
DisableSafeMode = 1 // Prevents the game from using the lowest settings upon relaunch if the game crashes.
FastTransitions = 1 // Increases speed of all transitions.
CreateLocalFix = 1 // For anti-aliasing support.
FPSLimit = 30 // Controls the framerate limit, but may cause bugs if increased.
PS2CameraSpeed = 1 // Changes camera speed to match PS2 version. Use with FPSLimit = 30.
GamepadControlsFix = 1 // Allows the same button to be assigned to multiple actions.
LightingFix = 1 // Fixes a flashlight bug, and restores the fog and lighting conditions for one of the ending cutscenes.
ReduceCutsceneFOV = 1 // Cutscenes will use vert- scaling to hide off-screen characters.
SteamCrashFix = 0 // Prevents a crash when Steam controller configuration is used.
IncreaseNoiseEffectRes = 512 // Use 128 and above to increase noise effect resolution.
FullscreenImages = 1 // Crops letterboxing to fill the screen.

// Nemesis2000 Fog Fix
[FOG]
fog_custom_on = 1 // 0 - Off, 1 - On
fog_transparency_layer1 = 128 // original value 128
fog_transparency_layer2 = 138 // original value 128
fog_layer1_x1 = 0.125 // original value 0.125
fog_layer1_x2 = 0.125 // original value 0.125
fog_layer1_y1 = 0.125 // original value 0.125
fog_layer1_y2 = 0.125 // original value 0.125
fog_layer2_density_mult = 1.0 // original value 1.0
fog_layer2_density_add = 50.0 // original value 0.0
fog_layer2_complexity = 0.066406 // original value 0.048828

// Update settings for modupdater
[MISC]
SelfUpdate = 0

[MODS]
d3d8.dll = https://github.com/elishacloud/Silent-Hill-2-Enhancements

[d3d8.dll]
PlaceToRoot

**Comment by Polymega** [comment]
Hi @retconn , 

Can you please share your `d3d8.log` file and also take a snapshot of your game's installation directory here? 

Thanks

**Comment by retconn** [comment]
I put all my settings from d3d8 in the original post but i think this is the log file and I will upload a png of the directory. Thank you.

d3d8 log:
![sh2 directory](https://user-images.githubusercontent.com/44321681/47810622-22f26800-dd1a-11e8-9c06-2da5fcf81b3f.png)

15:24:45.319 Starting Silent Hill 2 Enhancements! v1.2.1260.0
15:24:45.319 Dell Inc. Inspiron 5675 1.3.6 (Desktop)
15:24:45.319 Dell Inc. 07PR60 A00 (Desktop)
15:24:45.319 AMD Radeon (TM) RX 580
15:24:45.320 Windows 10 64-bit (10.0.2600)
15:24:45.320 Running from:  C:\Program Files (x86)\Mr DJ\Silent Hill 2\sh2pc.exe
15:24:45.320 Reading config file:  C:\Program Files (x86)\Mr DJ\Silent Hill 2\d3d8.ini
15:24:45.420 Loading as dll: d3d8.dll
15:24:45.422 Wrapper created for d3d8.dll
15:24:45.763 Hooking the FileSystem APIs...
15:24:45.766 DisableCDCheck Error: Could not find CD check function address in memory!
15:24:46.540 Updating SFX memory addresses...
15:24:46.544 Updating sddata.bin pointer memory addresses...
15:24:46.545 Setting PS2 Style Noise Filter...
15:24:46.547 Increasing the Draw Distance...
15:24:46.660 Setting Cemetery Lighting Fix...
15:24:46.661 Setting Rowboat Animation Fix...
15:24:46.662 Updating the Catacomb Meat Cold Rooms color...
15:24:46.666 Disabling Red Cross health indicator during cutscenes...
15:24:46.666 Loading the Nemesis2000 Fog Fix module...
15:24:46.668 Loading the WidescreenFixesPack and sh2proxy module...
15:24:48.491 Loading the modupdater module...
15:24:49.053 Redirecting 'Direct3DCreate8' ---> Passing on to 'Direct3DCreate9'
15:26:38.785 Unhooking library functions
15:26:38.786 Reseting screen resolution
15:26:39.410 Unloading Silent Hill 2 Enhancements!

**Comment by Polymega** [comment]
Hi @retconn ,

First, since you are using an AMD card, remove the WineD3D files from your directory (`d3d9.dll`, `libwine.dll`, and `wined3d.dll`).

Secondly, you have some files in your directory I've never seen/heard of and they may very well be conflicting with the mods for the Enhanced Edition. 

For example: What is the `Silent Hill 2 HD Patch` file? (Side note: I'd recommend [enabling showing known file extensions](https://www.thewindowsclub.com/show-file-extensions-in-windows) for future convenience.) What does the `SH2` batch file do for you?

Also, what is `FirewallInstallHelper.dll` and `GameuxInstallHelper.dll` do for you? Try removing these four files from the game's directory.

**Comment by AeroWidescreen** [comment]
> Silent Hill 2 HD Patch

Could be a resolution patcher, which is completely redundant of course. If you look at sh2pc.exe you can see its modified date does not match either of the exes you've provided in your guide, so that's likely part of the problem.

As someone with an RX580 and Windows 10, he clearly shouldn't be having any issues

**Comment by Polymega** [comment]
>  If you look at sh2pc.exe you can see its modified date does not match either of the exes you've provided in your guide, so that's likely part of the problem.

Nice catch. @retconn is your copy from a physical copy you own or did you obtain the game elsewhere? I'm not here to judge or anything; but it'll be helpful for us to know if you found a "packaged" download of all the mods that are including redundant/unnecessary files that are conflicting with the work from the EE team.

**Comment by retconn** [comment]
Forced windowed mode is still not working? 

Thank you. I removed the files you mentioned. I did get to one of the scenes that gave me lag and its not there anymore. I did get some small lag every few seconds (3s-10s) lag wandering the outside street areas but not sure if that is normal for loading the area.

**Comment by Polymega** [comment]
Where did you find/acquire all those extra files?

> I did get some small lag every few seconds (3s-10s) lag wandering the outside street areas but not sure if that is normal for loading the area.

Tone down the intensity or fully disable the Fog Fix to alleviate this problem.

**Comment by Polymega** [comment]
> is a vanilla bug if it is what I think it is. Something to do with the Advanced Filters option, which you probably don't want to disable.

Funny you mention this. I have a fix for this (altering the alphas on the glass windows) but I also have some hex addresses that will keep this room looking "correct" even when Advanced Filters is turned off. (Turn it on/off and pay attention to the white area outside, or simply pause the game (Esc) while in this room when Advanced Filters is on to see the room with AF turned off). 

I was holding off on posting about this one until things like the flashlight fix is at completion so as not to overwhelm. Mostly because this room is missing another feature (or it's severely toned down) like the motion blur and DoF: A pseudo/fake bloom effect.

**Comment by retconn** [comment]
i got the original package from a friend. i thought it was the hd mod and then i saw the recent updates so when i applied the recent updates i had all the issues i explained. but seems like most of it is resolved with the lag. 

the forced windowed mode still isnt working. the solution on my monitor is 1920x1080 and I lowered the resolution in the game and tried to restart it as well and I'm not getting it in windowed mode.

**Comment by AeroWidescreen** [comment]
> the forced windowed mode still isnt working. the solution on my monitor is 1920x1080 and I lowered the resolution in the game and tried to restart it as well and I'm not getting it in windowed mode.


>You can set the resolution in the ini file.

**Comment by Polymega** [comment]
> I lowered the resolution in the game and tried to restart it as well and I'm not getting it in windowed mode.

To expand upon Aero's solution: You will ignore whatever the resolution says _in-game_ because we are **overriding** that resolution through the `d3d8.ini` file. Whatever you set the resolution to in the .ini file (`ResX` and `ResY`) is what the game will play as.

If you want to play in windowed mode you need to enable `EnableWndMode` in the .ini file. You'll need to set `ResX` and `ResY` to something _smaller_ than your monitor's resolution to make it readily apparent that windowed mode is working.

If you want to play in windowed mode _with a border_ (so you can, for example move the windowed game around) you also need to enable `WndModeBorder` and ensure the height (`ResY`) of your custom resolution **is 32 pixels less** than your monitor's height, otherwise the border won't activate for the windowed game. So, for example, if your display is 1920 x 1080 you'll need to make sure the windowed game's height is _no more than_ 1048.

To recap: Make sure `ResX`, `ResY`, `EnableWndMode`, and `WndModeBorder` are set/activated in the `d3d8.ini` file.

**Comment by retconn** [comment]
Thank you I see now the resolution settings in the config file and I was able to get it in windowed mode now. Sorry about that.

One last thing really is if there is a way to have a free mouse on your PC while playing the game? When streaming it is very valuable to be able to have a mouse available and the game not go mute or having to tab away from the game to gain control over the mouse? If not, can this be somehow added functionality? Thank you. 

(Also seems like the textures are not glitching like I saw before such as in the apartment cutscene with Angela).

**Comment by Polymega** [comment]
> One last thing really is if there is a way to have a free mouse on your PC while playing the game?

No, this game came out before a time when Steam or GoG really took off and before playing windowed was a hot commodity for purposes such as streaming.
